kernel/src/log_segment/tests.rs

```diff
@@ -107,6 +118,100 @@ fn build_log_with_paths_and_checkpoint(
     (Box::new(client), log_root)
 }
 
+// Create an in-memory store and return the store and the URL for the store's _delta_log directory.
+fn new_in_memory_store() -> (Arc<InMemory>, Url) {
+    (
+        Arc::new(InMemory::new()),
+        Url::parse("memory:///")
+            .unwrap()
+            .join("_delta_log/")
+            .unwrap(),
+    )
+}
+
+// Writes a record batch obtained from engine data to the in-memory store at a given path.
+fn write_record_batch_to_store(
+    store: &Arc<InMemory>,
+    path: String,
+    data: Box<dyn EngineData>,
+) -> DeltaResult<()> {
+    let batch = ArrowEngineData::try_from_engine_data(data)?;
+    let record_batch = batch.record_batch();
+
+    let mut buffer = vec![];
+    let mut writer = ArrowWriter::try_new(&mut buffer, record_batch.schema(), None)?;
+    writer.write(record_batch)?;
+    writer.close()?;
+    println!("Writing to path: {}", path);
+
+    tokio::runtime::Runtime::new()
+        .expect("create tokio runtime")
+        .block_on(async {
+            if let Err(e) = store.put(&Path::from(path), buffer.into()).await {
+                eprintln!("Error writing to store: {}", e);
+            }
+        });
+
+    Ok(())
+}
+
+/// Writes all actions to a _delta_log parquet checkpoint file in the store.
+/// This function formats the provided filename into the _delta_log directory.
+fn add_checkpoint_to_store(
+    store: &Arc<InMemory>,
+    data: Box<dyn EngineData>,
+    filename: &str,
+) -> DeltaResult<()> {
+    let path = format!("_delta_log/{}", filename);
+    write_record_batch_to_store(store, path, data)
+}
+
+/// Writes all actions to a _delta_log/_sidecars file in the store.
+/// This function formats the provided filename into the _sidecars subdirectory.
+fn add_sidecar_to_store(
+    store: &Arc<InMemory>,
+    data: Box<dyn EngineData>,
+    filename: &str,
+) -> DeltaResult<()> {
+    let path = format!("_delta_log/_sidecars/{}", filename);
+    write_record_batch_to_store(store, path, data)
+}
+
+/// Writes all actions to a _delta_log json checkpoint file in the store.
+/// This function formats the provided filename into the _delta_log directory.
+fn add_json_checkpoint_to_store(
```

Comment:
I also think that this function could be used for commits, so maybe keep it consistent with the other naming pitch and have this be called `write_json_to_store`.